pyvortex/src/dataset.rs

```diff
@@ -128,17 +127,22 @@ impl TokioFileDataset {
         row_filter: Option<&Bound<'_, PyExpr>>,
         indices: Option<&PyArray>,
     ) -> PyResult<PyObject> {
-        let layout_reader = layout_stream_from_reader(
-            self_.file.clone(),
-            projection_from_python(columns)?,
-            row_filter_from_python(row_filter),
-            indices.map(PyArray::unwrap).cloned(),
-        )
-        .await?;
+        let mut scan = Scan::new(projection_from_python(columns)?);
+
+        if let Some(filter) = filter_from_python(row_filter) {
+            scan = scan.with_filter(filter);
+        }
+
+        if let Some(indices) = indices.map(PyArray::unwrap).cloned() {
+            let indices = Buffer::from_iter(indices.into_primitive()?.as_slice::<u64>().to_vec());
+            scan = scan.with_row_indices(indices);
+        }
+
+        let stream = self_.vxf.scan(scan)?;
```

Comment:
Seems like we should be able to share liens 130-141  with `read_array_from_reader`?

Reply:
I've inlined it because the types/different bounds messed up some things here, I can give it another go